packages/engine-frontend/components/AGConnectionPortal.tsx

```diff
@@ -8,23 +8,66 @@ import {cn} from '@openint/ui/utils'
 import {R} from '@openint/util'
 import {WithConnectConfig} from '../hocs/WithConnectConfig'
 import {_trpcReact} from '../providers/TRPCProvider'
-import {ConnectButton} from './ConnectButton'
 import {ResourceDropdownMenu} from './ResourceDropdownMenu'
+import React from 'react'
+import {VERTICAL_BY_KEY} from '@openint/cdk'
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from '@openint/ui'
+import type { ConnectorConfigFilters } from '../hocs/WithConnectConfig'
+import {IntegrationSearch} from './IntegrationSearch'
+
 
 type ConnectEventType = 'open' | 'close' | 'error'
 
 export interface AGConnectionPortalProps extends UIPropsNoChildren {
   onEvent?: (event: {type: ConnectEventType; ccfgId: Id['ccfg']}) => void
 }
 
-// TODO: Wrap this in memo so it does not re-render as much as possible.
-// Also it would be nice if there was an easy way to automatically prefetch on the server side
-// based on calls to useQuery so it doesn't need to be separately handled again on the client...
-export function AGConnectionPortal({
+// Custom comparison function for React.memo
+const areEqual = (prevProps: AGConnectionPortalProps, nextProps: AGConnectionPortalProps) => {
```

Comment:
there ougtta be something we can use for this instead of needing to write our own.